src/Microsoft.ML.StaticPipe/TransformsStatic.cs

```diff
@@ -1607,21 +1607,21 @@ private sealed class Reconciler : EstimatorReconciler
         /// speciﬁed shift-invariant kernel. With this transform, we are able to use linear methods (which are scalable) to approximate more complex kernel SVM models.
         /// </summary>
         /// <param name="input">The column to apply Random Fourier transfomration.</param>
-        /// <param name="dimension">The number of random Fourier features to create.</param>
+        /// <param name="rank">The number of random Fourier features to create.</param>
         /// <param name="useCosAndSinBases">If <see langword="true"/>, use both of cos and sin basis functions to create two features for every random Fourier frequency.
         /// Otherwise, only cos bases would be used.</param>
         /// <param name="generator">Which kernel to use. (if it is null, <see cref="GaussianKernel"/> is used.)</param>
         /// <param name="seed">The seed of the random number generator for generating the new features. If not specified global random would be used.</param>
-        public static Vector<float> LowerVectorSizeWithRandomFourierTransformation(this Vector<float> input,
-            int dimension = RandomFourierKernelMappingEstimator.Defaults.Rank, bool useCosAndSinBases = RandomFourierKernelMappingEstimator.Defaults.UseCosAndSinBases,
+        public static Vector<float> RandomFourierKernelMapper(this Vector<float> input,
+            int rank = RandomFourierKernelMappingEstimator.Defaults.Rank, bool useCosAndSinBases = RandomFourierKernelMappingEstimator.Defaults.UseCosAndSinBases,
             KernelBase generator = null, int? seed = null)
         {
             Contracts.CheckValue(input, nameof(input));
-            return new ImplVector<string>(input, new Config(dimension, useCosAndSinBases, generator, seed));
+            return new ImplVector<string>(input, new Config(rank, useCosAndSinBases, generator, seed));
         }
     }
 
-    public static class PcaEstimatorExtensions
```

Comment:
>PcaEstimatorExtensions [](start = 24, length = 22)

keeper

Reply:
What? You mean to have PcaStaticExtensions?

---
In reply to: [264837387](https://github.com/dotnet/machinelearning/pull/2888#discussion_r264837387) [](ancestors = 264837387)

Reply:
yes, thanks. 

---
In reply to: [264918309](https://github.com/dotnet/machinelearning/pull/2888#discussion_r264918309) [](ancestors = 264918309,264837387)